### DISCUSSION TITLE
Omnivore import

### DISCUSSION THREAD

**Discussion by Munchotaur** [wallabag/wallabag#7742] [Ideas]
Hi, now that Omnivore is merging/has been bought by ElevenLabs and turned into some AI-ish thing, is there a way to import Omnivore data?

As far as I can see, the meaningful part of the exported data is a numbered set of JSON metadata files (around 20 entries per file), along these lines:

```JSON
 [ {
    "id": "c8edb2b0-4d78-4c69-a09b-44ac5d50a114",
    "slug": "https-deslivresencommuns-org-1927109fd84",
    "title": "Des livres en communs",
    "description": "Pourquoi « libérer le livre » ? Voir notre présentation\nComment proposer votre projet ? Lire le règlement\nQu’est-ce qu’on publie ? Découvrir comment ça marche\n\n\nDes Livres en Communs, un …",
    "author": "Framasoft",
    "url": "https://deslivresencommuns.org",
    "state": "Active",
    "readingProgress": 0,
    "thumbnail": "https://deslivresencommuns.org/img/DLEC-logo.png",
    "labels": [],
    "savedAt": "2024-10-09T11:28:56.738Z",
    "updatedAt": "2024-10-09T11:28:57.050Z",
    "publishedAt": null
  },
...
```
There's also a subdirectory for the simplified HTML of articles, as rendered by Omnivore, and another subdirectory called 'highlights' containing markdown files, all of which are empty in my case!

**Comment by cb3inco** [comment]
I'm hoping for this too, but there isn't a ton of time if they need to build something.

**Reply by elzik** [reply]
No, the fact that all data will be deleted in a couple of weeks doesn't matter. If you [export and download your data](https://docs.omnivore.app/using/exporting.html) into the format that @Munchotaur suggests then it is yours to keep until there is an import feature for these metadata files.

**Reply by thiswillbeyourgithub** [reply]
According to the people of readeck there is more information from the API than the file export. Can't say what exactly myself

**Reply by thiswillbeyourgithub** [reply]
Addendum: that's probably why the suggested export is via obsidian as it uses the api?

Source for readeck: https://codeberg.org/readeck/readeck/issues/274

**Comment by nicosomb** [comment]
Thank you so much @Munchotaur for your post, we'll work on the import soon.

**Reply by Munchotaur** [reply]
I look forward to trying it out and will let you know if there are any issues. Do you need some sample JSON files from Omnivore to work on?

**Reply by nanotubing** [reply]
I can also supply an Omnivore export in case it helps testing

**Comment by nicosomb** [comment]
Blog post about omnivore import in progress https://github.com/wallabag/website/pull/68

**Comment by nicosomb** [comment]
A draft is available here: https://github.com/wallabag/wallabag/pull/7754

**Comment by QuAzI** [comment]
Now I got a mail about import feature... but import 41 files by hands makes me a bit unhappy)
So I used [jq](/jqlang/jq) to merge them into one (to rule them all...)
```shell
$ jq -sr 'map(.[])' metadata_*.json > combined.json
```

**Reply by nicosomb** [reply]
Wow, thank you so much. About the email you received, I can't write this tip in it because the wallabag.it customers are not all tech friendly 😉
